### PR TITLE
feat(tests): Add comprehensive test suite for the Project API

### LINKED ISSUE #7229: [feature]: [tests] Proposal to add test coverage for backend API endpoints
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Summary

During a deep-dive to understand the codebase, I noticed that the backend test suite has an architecture in place but contains several empty test files for core API modules, such as test_issue.py and test_project.py. This proposal is to begin methodically adding test coverage to these modules, following the established "outside-in" testing philosophy and leveraging the existing base classes like AuthenticatedAPITest to ensure consistency.

### Why should this be worked on?

Bolstering the test suite for these core features is a high-impact way to improve the project's long-term health. By starting to pay down this testing-related technical debt, we can make the entire application more stable and robust. As a contributor looking to help, I would appreciate any guidance on which API endpoints would be the most valuable to start with. I'm ready to begin with a small, focused Pull Request on a single endpoint to demonstrate the approach.

### PR DESCRIPTION
**Pull request by mohamedhayballa22**
### Description
This pull request introduces a comprehensive API contract test suite for the Project resource to improve the test coverage and long-term health of this component and the overall codebase.

This covers all CRUD operations (POST, GET, PATCH, DELETE) and includes extensive checks for:
- Core business logic and side-effects.
- Detailed permission enforcement for different user roles.
- Authentication requirements for all endpoints.
- Validation for invalid data and uniqueness constraints.
- Edge cases like retrieving non-existent or archived projects.

To support this and future tests, this PR also introduces a reusable workspace fixture in conftest.py.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Improvement (change that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Performance improvements
- [ ] Documentation update

### Test Scenarios 
The changes in this PR consist entirely of new tests. All new and existing tests in the suite were run via pytest and passed successfully in the local development environment. The new tests themselves serve as the verification for the existing Project API functionality.

### References
Closes #7229

## Summary by CodeRabbit

- **Tests**
  - Added new contract tests for Project API endpoints, covering project creation, retrieval, updating, and deletion within a workspace context.
  - Tests include scenarios for permissions, required and optional fields, conflict handling, and access control for different user roles.
  - Introduced a new fixture to streamline workspace setup in tests.

### CODE REVIEW

File: apiserver/plane/tests/conftest.py [hunk 1]
```diff
@@ -118,3 +120,29 @@ def plane_server(live_server):
     Returns a live Django server for testing HTTP requests.
     """
     return live_server
+
+
+@pytest.fixture
+def workspace(session_client):
```
Comment (dheeru0198) [inline apiserver/plane/tests/conftest.py]: @mohamedhayballa22 I suggest you to directly create the workspace for this fixture using the `Workspace` model instead of the API.

Comment (mohamedhayballa22) [reply apiserver/plane/tests/conftest.py]: @dheeru0198 Done!

**Review by dheeru0198 — APPROVED** [review]
